File: app/src/main/java/com/example/pam_app/db/IncomeDao.java
package com.example.pam_app.db;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Transaction;

import java.util.List;

import io.reactivex.Flowable;

@Dao
public interface IncomeDao {

    @Query("SELECT * FROM income")
    Flowable<List<IncomeEntity>> getList();

    @Insert
    void create(final IncomeEntity income);

    @Query("DELETE FROM income WHERE id=:id")
    void delete(final int id);

    @Transaction
    @Query("SELECT * FROM income WHERE id=:id")
    Flowable<IncomeEntity> get(final int id);
}

File: app/src/main/java/com/example/pam_app/db/IncomeDao.kt
package com.example.pam_app.db

import androidx.room.*
import io.reactivex.Flowable
import java.util.*

@Dao
interface IncomeDao {
    @get:Query("SELECT * FROM income")
    val list: Flowable<List<IncomeEntity?>?>

    @Insert
    fun create(income: IncomeEntity?): Long

    @Query("DELETE FROM income WHERE id=:id")
    fun delete(id: Int)

    @Transaction
    @Query("SELECT * FROM income WHERE id=:id")
    operator fun get(id: Int): Flowable<IncomeEntity?>

    @TypeConverters(DateConverter::class)
    @Query("SELECT * FROM income WHERE incomeType=:type AND date<:date")
    fun getList(type: Int, date: Date?): Flowable<List<IncomeEntity?>?>
}